csharp/src/Ice/CommunicatorI.cs

```diff
@@ -111,7 +112,7 @@ public ObjectAdapter createObjectAdapterWithRouter(string name, RouterPrx router
                 getProperties().setProperty(entry.Key, entry.Value);
             }
 
-            return _instance.objectAdapterFactory().createObjectAdapter(name, router);
+            return _instance.objectAdapterFactory().createObjectAdapter(name, router, authenticationOptions: null);
```

Comment:
Can you open an issue to make Router and Endpoints incompatible?

More generally Endpoints should be incompatible with any OA used for "bidir" communications.

Reply:
https://github.com/zeroc-ice/ice/issues/1986